src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/IPropertyModelBaseConvention.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationModels
+{
+    /// <summary>
+    /// Allows customization of the <see cref="PropertyModel"/>.
+    /// </summary>
+    /// <remarks>
+    /// To use this interface, create an <see cref="System.Attribute"/> class which implements the interface and
+    /// place it on a controller property.
+    ///
+    /// <see cref="IPropertyModelConvention"/> customizations run after
+    /// <see cref="IControllerModelConvention"/> customizations.
+    /// </remarks>
+    public interface IPropertyModelBaseConvention
+    {
+        /// <summary>
+        /// Called to apply the convention to the <see cref="PropertyModel"/>.
+        /// </summary>
+        /// <param name="parameter">The <see cref="PropertyModel"/>.</param>
+        void Apply(PropertyModelBase parameter);
```

Comment:
Do we need this? Could we make these all just be derived from `ParameterModelBase`? Properties are parameters

Reply:
You would't have access to the backing `PropertyInfo` \ `ParameterInfo` until you downcasted. You could use the `Attribute` collection to drive some of the behavior, but not all. Maybe we could mitigate some of it by making the base type have a `ParameterType` so you could make type based decisions in conventions.

Reply:
The base type should have a `MemberInfo`.

Reply:
Give a try, let's see if it looks reasonable or not.